types/Cargo.toml

```diff
@@ -0,0 +1,27 @@
+[package]
+name        = "cuprate-types"
+version     = "0.0.0"
+edition     = "2021"
+description = "Cuprate data types"
+license     = "MIT"
+authors     = ["hinto-janai"]
+repository  = "https://github.com/Cuprate/cuprate/tree/main/types"
+keywords    = ["cuprate", "types"]
+
+[features]
+default = ["service"]
+service = ["dep:strum"]
+
+[dependencies]
+borsh            = { workspace = true, optional = true }
+cfg-if           = { workspace = true }
+curve25519-dalek = { workspace = true }
+monero-serai     = { workspace = true }
+paste            = { workspace = true }
+thiserror        = { workspace = true }
+serde            = { workspace = true, optional = true }
+strum            = { workspace = true, features = ["derive"], optional = true }
```

Comment:
what are your plans for this? 

Although I don't want to worry about it too much now, I think a goal should be to have as little dependencies as possible (within reason)

Reply:
Also I don't think all the dependencies defined here are actually used

Reply:
Nothing in particular, mostly just implementing now for future proofing if it's ever needed, should it be removed?

Reply:
Or feature gated?